controllers/runtimecomponent_controller.go

```diff
@@ -221,6 +221,17 @@ func (r *RuntimeComponentReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		}
 	}
 
+	// Check if the ServiceAccount has a valid pull secret before creating the deployment/statefulset
+	// or setting up knative. Otherwise the pods can go into an ImagePullBackOff loop
+	saName := instance.Name
+	if instance.Spec.ServiceAccountName != nil && *instance.Spec.ServiceAccountName != "" {
+		saName = *instance.Spec.ServiceAccountName
+	}
+	saErr := appstacksutils.PullSecretExists(saName, instance.Namespace, r.GetClient())
```

Comment:
To minimize the amount of changes needed for each controller, it'll be good if the above logic to determine the service account name is also moved into the util method. The variable `ba` (of type common.BaseComponent) can be passed instead of _saName_ and _instance.Namespace_. They can be retrieved from `ba` along with CR fields (i.e. ba.GetServiceAccountName()). Util method [CustomizeRoute](https://github.com/application-stacks/runtime-component-operator/blob/db4d0420b902d713a0b3e0e2f6197c9addbb83a4/utils/utils.go#L96) is a good reference. It's fine to pass in client along with ba.

Suggested name for the method: ServiceAccountPullSecretExists